src/rules/variableNameRule.ts

```diff
@@ -60,130 +62,136 @@ export class Rule extends Lint.Rules.AbstractRule {
-            super.visitVariableStatement(node);
+        const { text } = name;
+        if (initializer && isAlias(text, initializer)) {
+            return;
         }
-    }
 
-    private isAlias(name: ts.Identifier, initializer: ts.Expression): boolean {
-        if (initializer.kind === ts.SyntaxKind.PropertyAccessExpression) {
-            return (initializer as ts.PropertyAccessExpression).name.text === name.text;
-        } else if (initializer.kind === ts.SyntaxKind.Identifier) {
-            return (initializer as ts.Identifier).text === name.text;
+        if (!isCamelCase(text, options) && !isUpperCase(text)) {
+            ctx.addFailureAtNode(name, Rule.FORMAT_FAILURE);
         }
-        return false;
     }
 
-    private handleVariableNameFormat(name: ts.Identifier, initializer?: ts.Expression) {
-        const variableName = name.text;
-
-        if (initializer && this.isAlias(name, initializer)) {
-            return;
-        }
-
-        if (this.shouldCheckFormat && !this.isCamelCase(variableName) && !isUpperCase(variableName)) {
-            this.addFailureAtNode(name, Rule.FORMAT_FAILURE);
+    function handleVariableNameKeyword(name: ts.Identifier): void {
+        if (options.banKeywords && BANNED_KEYWORDS.has(name.text)) {
+            ctx.addFailureAtNode(name, Rule.KEYWORD_FAILURE);
         }
     }
+}
 
-    private handleVariableNameKeyword(name: ts.Identifier) {
-        const variableName = name.text;
-
-        if (this.shouldBanKeywords && BANNED_KEYWORDS.indexOf(variableName) !== -1) {
-            this.addFailureAtNode(name, Rule.KEYWORD_FAILURE);
-        }
+function isAlias(name: string, initializer: ts.Expression): boolean {
+    switch (initializer.kind) {
+        case ts.SyntaxKind.PropertyAccessExpression:
+            return (initializer as ts.PropertyAccessExpression).name.text === name;
+        case ts.SyntaxKind.Identifier:
+            return (initializer as ts.Identifier).text === name;
+        default:
+            return false;
     }
+}
 
-    private isCamelCase(name: string) {
-        const firstCharacter = name.charAt(0);
-        const lastCharacter = name.charAt(name.length - 1);
-        const middle = name.substr(1, name.length - 2);
+function isCamelCase(name: string, options: Options): boolean {
+    const firstCharacter = name.charAt(0);
+    const lastCharacter = name.charAt(name.length - 1);
+    const middle = name.substr(1, name.length - 2);
 
-        if (name.length <= 0) {
-            return true;
-        }
-        if (!this.hasOption(OPTION_LEADING_UNDERSCORE) && firstCharacter === "_") {
-            return false;
-        }
-        if (!this.hasOption(OPTION_TRAILING_UNDERSCORE) && lastCharacter === "_") {
-            return false;
-        }
-        if (!this.hasOption(OPTION_ALLOW_PASCAL_CASE) && !isLowerCase(firstCharacter)) {
-            return false;
-        }
-        return middle.indexOf("_") === -1;
+    if (name.length <= 0) {
```

Comment:
I know that check was there before, but when would this ever be true?
Edit: function declarations without a name have an identifier with zero width, but that is invalid syntax anyway.